app/src/main/java/net/squanchy/onboarding/location/LocationOnboardingActivity.java

```diff
@@ -48,21 +50,25 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
         service = component.proximityService();
         navigator = component.navigator();
         proximityPreconditions = component.proximityPreconditions();
+        optInPersister = component.optInPersister();
 
         googleApiClient.connect();
 
         setContentView(R.layout.activity_location_onboarding);
         contentRoot = findViewById(R.id.onboarding_content_root);
 
-        findViewById(R.id.skip_button).setOnClickListener(view -> markPageAsSeenAndFinish());
+        findViewById(R.id.skip_button).setOnClickListener(view -> {
+            optInPersister.setOptInPreferenceTo(false);
```

Comment:
Extract a method here please for these two calls? something like `optOutFromProximity` so it mirror the other case maybe even